src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,202 @@
+        } else if (index == size - 1) {
+            deleteElement = unlinkLast();
+        } else {
+            deleteElement = unlink(searchNodeByIndex(index));
+        }
+        return deleteElement;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = searchNodeByElement(object);
+        if (currentNode == null) {
+            return false;
+        } else if (currentNode.equals(head)) {
+            unlinkFirst();
+            return true;
+        } else if (currentNode.equals(tail)) {
+            unlinkLast();
+            return true;
+        }
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    private void addFirst(T value) {
+        Node<T> newNode = new Node<T>(null, value, null);
+        head = newNode;
+        tail = head;
+    }
+
+    private void insertBefore(T value) {
+        Node<T> newNode = new Node<>(null, value, head);
+        head.prev = newNode;
+        head = newNode;
+    }
+
+    private void insertAfter(T value) {
+        Node<T> newNode = new Node<T>(tail, value, null);
+        tail.next = newNode;
+        tail = newNode;
+    }
+
+    private T unlink(Node<T> object) {
+        Node<T> currentNode = object;
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return currentNode.item;
+    }
+
+    private T unlinkFirst() {
+        T deleteElement = null;
+        if (size == 1) {
+            deleteElement = head.item;
+            tail = null;
+            head = null;
+        } else {
+            deleteElement = head.item;
+            head = head.next;
+            head.prev = null;
+        }
+        size--;
+        return deleteElement;
+    }
+
+    private T unlinkLast() {
+        T deleteElement = null;
+        deleteElement = tail.item;
+        tail = tail.prev;
+        tail.next = null;
+        size--;
+        return deleteElement;
+    }
+
+    private Node<T> searchNodeByElement(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.item, object)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private Node<T> searchNodeByIndex(int index) {
+        if (checkIndex(index)) {
+            Node<T> currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode;
+        }
+        return null;
+    }
+
+    private boolean checkIndex(int index) {
```

Comment:
not fixed, name doesn't tell us boolean will be returned. Let's make it void
```suggestion
    private void checkIndex(int index) {
```